### PR TITLE
Break metal into smaller files

### LINKED ISSUE #8: Split Backbone metal into smaller files
It'd be great to use @jmeas's gulp build process to break this up into smaller files

### PR DESCRIPTION
**Pull request by paulfalgout**
Resolves #8 

A quick initial go at it.

Includes funny business around the `_noop` just to get it to build.

Seems to have introduced a regression related to deprecate there as well, but I'm not certain why.

I am moderately concerned particularly with the RegEx for `CONTAINS_SUPER` that the boilerplate is building it a little funny, and I mentioned that over on that commit.

Also should `_wrap` or `_wrapAll` be their own exports to clean up class?

Initial thoughts?

### CODE REVIEW

File: src/noop.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Empty no-op function.
+ *
+ * @private
+ * @method _noop
+ */
+const _noop = function() {};
+
+export default _noop;
```
Comment (jamiebuilds) [inline src/noop.js]: meh, just inline these.

File: src/backbone-metal.js [hunk 2]
```diff
@@ -1,529 +1,25 @@
-import _ from 'underscore';
 import Backbone from 'backbone';
 
+import Class from './class';
```
Comment (jamiebuilds) [inline src/backbone-metal.js:2]: Can you remove the newlines between your imports?

File: src/backbone-metal.js [hunk 3]
```diff
@@ -1,529 +1,25 @@
...
+import Mixin from './mixin';
+import Err from './error';
+import deprecate from './deprecate';
+import Events from './events';
+
 /**
  * @module Metal
  */
 const Metal = Backbone.Metal = {};
```
Comment (jamiebuilds) [inline src/backbone-metal.js]:
``` js
export default Backbone.Metal = {
  Class,
  Mixin,
  Error: Err,
  deprecate,
  Events,
};
```

**Comment by jamiebuilds** [comment]
To fix the build you need to fix this: https://github.com/marionettejs/backbone-metal/blob/master/test/unit/support.js#L10

**Comment by paulfalgout** [comment]
Ok.  Cleaned up.

File: src/backbone-metal.js [hunk 4]
```diff
@@ -1,534 +1,23 @@
...
-
-/**
- * A safe reference to the console.warn method that will fallback a noop.
- *
- * @private
- * @property _warnMethod
+ * @mixes Events
  */
-const _warnMethod = _console.warn || _console.log || _noop;
+Class.mixin(Events);
```
Comment (jamiebuilds) [inline src/backbone-metal.js]: Is there any issue doing this within the `class.js` file?

Comment (paulfalgout) [inline src/backbone-metal.js]: nope.  moved.

**Comment by jamiebuilds** [comment]
Cool

**Pull request merged**